### PR TITLE
Fix attach_interface on s390x

### PR DESCRIPTION
**Pull request by smitterl**
On s390x interfaces have address types ccw which don't allow for
rom tuning.

1. get_formatted_iface_dict:
 a. Moved list creation to own function to
improve readability.
 b. Add rom tuning only if guest is not s390x
2. logging.info/params_format:
 a. Moved logging before and after calling function into function
to improve readability

Tests were run both on x86_64 and s390x, libvirt 4.5.0/qemu-kvm 2.12.0
x86_64:
```bash
# avocado run --vt-type libvirt --vt-connect-uri qemu:///system virsh.attach_detach_interface.normal_test.vm_running.direct.at_detach_device.domname
JOB ID     : b2598690367eab841cbb19a01515e740990858b0
JOB LOG    : /root/avocado/job-results/job-2019-10-24T19.19-b259869/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.normal_test.vm_running.direct.at_detach_device.domname: PASS (30.12 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 31.72 s
```
s390x:
```bash
# avocado run --vt-type libvirt --vt-machine-type s390-virtio --vt-connect-uri qemu:///system virsh.attach_detach_interface.normal_test.vm_running.direct.at_detach_device.domname
JOB ID     : 0a33c7b98ddc3ba7f5144ad8e069862d6c56b1cb
JOB LOG    : /root/avocado/job-results/job-2019-10-24T13.11-0a33c7b/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.normal_test.vm_running.direct.at_detach_device.domname: PASS (117.77 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 119.29 s
```

### CODE REVIEW

**Comment by smitterl** [comment]
@dzhengfy :)

File: libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface.py [hunk 1]
```diff
@@ -419,3 +405,25 @@ def run(test, params, env):
         if vm.is_alive():
             vm.destroy()
         backup_xml.sync()
+
+
+def get_formatted_iface_dict(names, vm_arch_name):
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface.py]: miss doc description

File: libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface.py [hunk 2]
```diff
@@ -419,3 +405,25 @@ def run(test, params, env):
...
+        "inbound", "outbound", "link", "target", "mac", "source",
+        "boot", "backend", "type", "mode"
+    ]
+    # For s390-virtio interface addresses are of type ccw
+    # rom tuning is only allowed for type pci
+    if not vm_arch_name == "s390x":
+        update_list.append("rom")
+
+    for update_item in update_list:
+        if names["iface_" + update_item]:
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface.py]: line 425 to 428 may be simplified as below:
iface_dict.update({update_item: names.get("iface_" + update_item)})

Comment (smitterl) [reply libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface.py]: Thank you! :D

**Comment by smitterl** [comment]
@chunfuwen Thanks for feedback. I fixed the issues you pointed out and run a single test:
```bash
# avocado run --vt-type libvirt --vt-machine-type i440fx --vt-connect-uri qemu:///system virsh.attach_detach_interface.normal_test.vm_paused.default_network.domname
JOB ID     : a17a576ce15bf43d99797c6eaeee5fa487ddb309
JOB LOG    : /root/avocado/job-results/job-2019-10-28T11.05-a17a576/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.normal_test.vm_paused.default_network.domname: PASS (27.56 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 29.08 s
```

**Pull request merged**